dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavTool.java

```diff
@@ -0,0 +1,301 @@
+                                .getInode();
+            } else {
+                parentId = null;
+            }
+            result = new NavResult(parentId, host.getIdentifier(), folder.getInode(), languageId);
+
+            Identifier ident = APILocator.getIdentifierAPI()
+                .find(folder);
+            result.setHref(ident.getURI());
+            result.setTitle(folder.getTitle());
+            result.setOrder(folder.getSortOrder());
+            result.setType("folder");
+            result.setPermissionId(folder.getPermissionId());
+            List<NavResult> children = new ArrayList<NavResult>();
+            List<String> folderIds = new ArrayList<String>();
+            result.setChildren(children);
+            result.setChildrenFolderIds(folderIds);
+            result.setShowOnMenu(folder.isShowOnMenu());
+
+            List menuItems;
+            if (path.equals("/"))
+                menuItems = APILocator.getFolderAPI()
+                    .findSubFolders(host, true);
+            else
+                menuItems = APILocator.getFolderAPI()
+                    .findMenuItems(folder, systemUserParam, true);
+
+            for (Object item : menuItems) {
+                if (item instanceof Folder) {
+                    Folder itemFolder = (Folder) item;
+                    ident = APILocator.getIdentifierAPI()
+                        .find(itemFolder);
+                    NavResult nav = new NavResult(folder.getInode(), host.getIdentifier(), itemFolder.getInode(), languageId);
+
+                    nav.setTitle(itemFolder.getTitle());
+                    nav.setHref(ident.getURI());
+                    nav.setOrder(itemFolder.getSortOrder());
+                    nav.setType("folder");
+                    nav.setPermissionId(itemFolder.getPermissionId());
+                    nav.setShowOnMenu(itemFolder.isShowOnMenu());
+
+                    // it will load lazily its children
+                    folderIds.add(itemFolder.getInode());
+                    nav = new NavResultHydrated(nav, this.context);
+                    children.add(nav);
+                } else if (item instanceof IHTMLPage) {
+                    IHTMLPage itemPage = (IHTMLPage) item;
+
+                    if (itemPage.getLanguageId() == languageId || LanguageWebAPI.canDefaultPageToDefaultLanguage()) {
+                        final String httpProtocol = "http://";
+                        final String httpsProtocol = "https://";
+
+                        ident = APILocator.getIdentifierAPI()
+                            .find(itemPage);
+
+                        String redirectUri = itemPage.getRedirect();
+                        NavResult nav =  new NavResult(folder.getInode(), host.getIdentifier(), languageId);
+
+                        nav.setTitle(itemPage.getTitle());
+                        if (UtilMethods.isSet(redirectUri) && !redirectUri.startsWith("/")) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)